url4fdf79ae2c_beckchr_juel_tgz-pJ8-de_odysseus_el_samples_xml_sax_XMELFilterJ8/src/de/odysseus/el/tree/impl/ast/AstFunction.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2006-2009 Odysseus Software GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.odysseus.el.tree.impl.ast;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import javax.el.ELContext;
+import javax.el.ELException;
+
+import de.odysseus.el.misc.LocalMessages;
+import de.odysseus.el.tree.Bindings;
+import de.odysseus.el.tree.FunctionNode;
+
+public class AstFunction extends AstRightValue implements FunctionNode {
+	private final int index;
+	private final String name;
+	private final AstParameters params;
+	private final boolean varargs;
+
+	public AstFunction(String name, int index, AstParameters params) {
+		this(name, index, params, false);
+	}
+
+	public AstFunction(String name, int index, AstParameters params, boolean varargs) {
+		this.name = name;
+		this.index = index;
+		this.params = params;
+		this.varargs = varargs;
+	}
+
+	/**
+	 * Invoke method.
+	 * @param bindings
+	 * @param context
+	 * @param base
+	 * @param method
+	 * @return method result
+	 * @throws InvocationTargetException
+	 * @throws IllegalAccessException
+	 */
+	protected Object invoke(Bindings bindings, ELContext context, Object base, Method method)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.